### PR TITLE
Minor enhancements to members management

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
This PR  fixes a few minor problems in member management and makes some cosmetic changes.
- create errors with new and remove the error code
- read the project from the cache and adapt the test
- add all service accounts to the subjectList (only items with index > 0 are project members)
- split the `members` service into multiple files (like terminals)
- remove unused methods
- add a few unit tests for some uncovered edge cases

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
The changes can be tested with:
```
docker run -it --rm -e LOG_LEVEL=trace -e KUBECONFIG="/home/node/path/from/home/to/kubeconfig" \
  -v $HOME:/home/node -p 8080:8080 eu.gcr.io/gardener-project/gardener/dashboard:1.47.0-dev-cb8afd63 \ 
  /home/node/.gardener/config.yaml
``` 
**Release note**:

```improvement user

```

### CODE REVIEW

File: backend/lib/services/members/MemberManager.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+const SubjectListItem = require('./SubjectListItem')
+const SubjectList = require('./SubjectList')
+
+class MemberManager {
+  constructor (client, userId, project, serviceAccounts) {
+    this.client = client
+    this.userId = userId
+    this.namespace = project.spec.namespace
+    this.projectName = project.metadata.name
+    const members = _.cloneDeep(project.spec.members)
```
Comment (holgerkoser) [inline backend/lib/services/members/MemberManager.js:36]: Not needed

File: backend/lib/services/members/MemberManager.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    }
+    return item.member
+  }
+
+  async create (id, { roles, description }) {
+    let item = this.subjectList.get(id)
+    if (item) {
+      throw new Conflict(`${item.kind} '${id}' already exists`)
+    }
+    item = SubjectListItem.create(id, Number.MAX_SAFE_INTEGER)
```
Comment (holgerkoser) [inline backend/lib/services/members/MemberManager.js:64]: Create Constant for -1 and MAX_INT

File: backend/test/services.members.spec.js [hunk 3]
```diff
@@ -120,7 +127,8 @@ describe('services', function () {
           name: 'robot-sa',
           namespace: 'garden-foo',
           annotations: {
-            'garden.sapcloud.io/createdBy': 'foo'
+            'garden.sapcloud.io/createdBy': 'foo',
```
Comment (holgerkoser) [inline backend/test/services.members.spec.js:130]: Change to new

File: backend/lib/services/members/MemberManager.js (hunk 1 above)
Comment (holgerkoser) [reply backend/lib/services/members/MemberManager.js:36]: ✔️

File: backend/lib/services/members/MemberManager.js (hunk 2 above)
Comment (holgerkoser) [reply backend/lib/services/members/MemberManager.js:64]: ✔️

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm
